Source/bmalloc/libpas/src/libpas/pas_local_allocator_inlines.h

```diff
@@ -1074,17 +1074,27 @@ pas_local_allocator_refill_with_known_config(
                          anymore. */
         if (view_cache_result.did_succeed) {
             pas_local_view_cache* view_cache;
+            bool can_pop;
             
             view_cache = (pas_local_view_cache*)view_cache_result.allocator;
             
             PAS_TESTING_ASSERT(!view_cache->scavenger_data.is_in_use);
             
             view_cache->scavenger_data.is_in_use = true;
             pas_compiler_fence();
+
+            /* Even after setting is_in_use = true, it is possible that this bit gets false and decommitted.
+               Just before setting it, scavenger runs and makes this cache stopped state. If the cache is the stopped state,
+               it is possible that scavenger further runs and makes this cache even decommitted since is_in_use can prevent
+               changing active to stopped state, but does not prevent changing stopped state to decommitted state. In that case,
+               even just after setting is_in_use, it is possible that this becomes false since cache is already decommitted.
+               pas_local_view_cache_prepare_to_pop ensures that view_cache is committed. */
+            can_pop = pas_local_view_cache_prepare_to_pop(view_cache);
```

Comment:
If the view is decommitted / stopped, we revive it in `pas_local_view_cache_prepare_to_pop`, and it requires taking a scavenger_lock. So `pas_local_allocator_scavenger_data_did_use_for_allocation` is ensured to be happen after `pas_local_view_cache_prepare_to_pop`.
If it is not revived, then, ordering does not matter since scavenger cannot make it stopped / decommitted.